clang/test/Sema/builtin-counted-by-ref.c

```diff
@@ -0,0 +1,102 @@
+// RUN: %clang_cc1 -std=c99 -fsyntax-only -verify %s
+
+typedef unsigned long int size_t;
+
+int global_array[42];
+int global_int;
+
+struct fam_struct {
```

Comment:
Another kind of FAM struct we should probably test is:
```
struct erp {
  int count;
  int fam[][12]; __attribute__((counted_by(count)));
};
```

Reply:
That's not a valid flexible array member, so the attribute should be rejected from it.